src/dma.rs

```diff
@@ -55,7 +57,9 @@ impl<B, C: Channel, T> Transfer<B, C, T> {
         B: ReadBuffer + 'static,
         T: Target<C>,
     {
-        let (ptr, len) = buffer.read_buffer();
+        // NOTE(unsafe) cannot call `&mut self` methods on `buffer` because its
+        // concrete type is unknown here
```

Comment:
```suggestion
        // NOTE(unsafe) We can't (and won't) call `&mut self` methods on
        // `buffer` after this call and until the end of the transfer, because
        // its concretetype is unknown here
```